### ISSUE TITLE
[macOS] 'presentCodeRedemptionSheet' has been explicitly marked unavailable here

### ISSUE THREAD

**Issue by EliasDeuss** [RevenueCat/purchases-flutter#150]
## Common project specific issues

**Describe the bug**
Happens in the build process for a MacOS app.

command:
```flutter run -d macOS```

Flutter Doctor -v
```
[✓] Flutter (Channel dev, 1.26.0-17.1.pre, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 1.26.0-17.1.pre at /Users/elias/flutter
    • Framework revision c40e5294dd (3 days ago), 2021-01-29 08:44:18 -0800
    • Engine revision 5d3477eab4
    • Dart version 2.12.0 (build 2.12.0-259.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1
```

full error
```
elias@eliass-mbp The-Unwanteds % flutter run -d macOS
Launching lib/main.dart on macOS in debug mode...
Running pod install...                                             11.3s
/Users/elias/flutter/.pub-cache/hosted/pub.dartlang.org/purchases_flutter-2.0.2/macos/Classes/PurchasesFlutterPlugin.m:320:32: error: 'presentCodeRedemptionSheet' is unavailable: not available on macOS
        [RCCommonFunctionality presentCodeRedemptionSheet];             
                               ^                                        
In module 'PurchasesHybridCommon' imported from /Users/elias/flutter/.pub-cache/hosted/pub.dartlang.org/purchases_flutter-2.0.2/macos/Classes/PurchasesFlutterPlugin.m:5:
/Users/elias/Documents/GitHub/The-Unwanteds/macos/Pods/PurchasesHybridCommon/ios/PurchasesHybridCommon/PurchasesHybridCommon/RCCommonFunctionality.h:60:1: note: 'presentCodeRedemptionSheet' has been explicitly marked unavailable here
+ (void)presentCodeRedemptionSheet API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(tvos, macos, watchos);
^                                                                       
1 error generated.                                                      
note: Using new build system                                            
note: Building targets in parallel                                      
note: Planning build                                                    
note: Constructing build description                                    
** BUILD FAILED **                                                      
                                                                        
Building macOS application...                                           
Exception: Build process failed
```

**Comment by aboedo** [comment]
hi! @EliasDeuss 👋 Thanks for reporting! 

My bad, I figured that [this condition](https://github.com/RevenueCat/purchases-flutter/blob/develop/ios/Classes/PurchasesFlutterPlugin.m#L319)
```objective-c
if (@available(iOS 14.0, *)) { 
```
would evaluate to false on macOS, but apparently it's actually true on macOS, it only guards against older versions of iOS. 
[The docs](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/marking_api_availability_in_objective-c) aren't very clear about this. 

I'll patch it. Thanks again for reporting!

**Comment by aboedo** [comment]
Addressed in #151, coming out soon

**Comment by aboedo** [comment]
@EliasDeuss `2.0.3` has been released with the fix. Closing this now. Thanks once more for reporting!
